### PR TITLE
csi: volumes listed in `nomad node status`

### LINKED ISSUE #6991: CSI:  CLI/API to support showing  volume info in nomad node status
Once volume mounting is implemented(TODO link PR), the CLI for `nomad node status` should show information about volumes on that node. 

Things to consider:
   - What information to show about volumes on that node
   - If we have fingerprinting/stats emission from used volumes, could show usage stats here

### PR DESCRIPTION
**Pull request by langmartin**
Fixes #6991 For consideration: we show `Schedulable` in the verbose status, which is maybe a bit confusing in this context?

### CODE REVIEW

File: nomad/state/state_store.go [hunk 1]
```diff
@@ -1710,6 +1710,55 @@ func (s *StateStore) CSIVolumesByIDPrefix(ws memdb.WatchSet, namespace, volumeID
...
+		return nil, fmt.Errorf("alloc lookup failed: %v", err)
+	}
+	snap, err := s.Snapshot()
+	if err != nil {
+		return nil, fmt.Errorf("snapshot failed: %v", err)
+	}
+
+	allocs, err = snap.DenormalizeAllocationSlice(allocs)
+	if err != nil {
+		return nil, fmt.Errorf("alloc denormalize failed: %v", err)
```
Comment (tgross) [inline nomad/state/state_store.go]: Is this error message meaningful to operators?

File: command/node_status.go [hunk 2]
```diff
@@ -515,6 +536,52 @@ func (c *NodeStatusCommand) outputNodeVolumeInfo(node *api.Node) {
...
+	volumes := map[string]*api.CSIVolumeListStub{}
+	vs, _ := client.Nodes().CSIVolumes(node.ID, nil)
+	for _, v := range vs {
+		n := volNames[v.ID]
+		volumes[n] = v
+	}
+
+	// Output the volumes in name order
+	output := make([]string, 0, len(names)+1)
+	output = append(output, "ID|Name|Plugin ID|Schedulable|Access Mode")
```
Comment (tgross) [inline command/node_status.go]: Do we always want to output `[bold]CSI Volumes` and this header, even if the length of volumes ends up being 0 by the time we're done querying?

File: command/node_status.go [hunk 3]
```diff
@@ -515,6 +536,52 @@ func (c *NodeStatusCommand) outputNodeVolumeInfo(node *api.Node) {
...
+	for _, name := range names {
+		v := volumes[name]
+		output = append(output, fmt.Sprintf(
+			"%s|%s|%s|%t|%s",
+			v.ID,
+			name,
+			v.PluginID,
+			v.Schedulable,
```
Comment (tgross) [inline command/node_status.go]: > For consideration: we show `Schedulable` in the verbose status, which is maybe a bit confusing in this context?

There might be value in having a consistent representation even if it's not super _useful_ in this context. Do we want to have the `Provider` here though?

File: command/node_status.go (hunk 2 above)
Comment (langmartin) [reply command/node_status.go]: We do, which matches some of the other sections. The verbose output is already pretty verbose, so it seems like we favor consistency.

File: command/node_status.go (hunk 3 above)
Comment (langmartin) [reply command/node_status.go]: whoops, fixed in 32e0ad21f

File: nomad/state/state_store.go (hunk 1 above)
Comment (langmartin) [reply nomad/state/state_store.go]: Fixed in 666a8a7f8

**Pull request merged**